noxfile.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import nox
+
+
+ALL_PYTHONS = ['3.7', '3.8', '3.9']
+
+nox.options.sessions = ['lint', 'type', 'tests']
+
+
+@nox.session(reuse_venv=True)
+def lint(session):
+    """
+    Run the linter.
+    """
+    session.install('pre-commit')
+    session.run('pre-commit', 'run', '--all-files', *session.posargs)
+
+
+@nox.session(python=ALL_PYTHONS, reuse_venv=True)
```

Comment:
We only have one Python version dependent, so I would prefer to inline the versions.

Reply:
I like having it at the top, makes it easier to notice & maintain (such as adding 3.10), but sure, if you prefer.